core/src/test/java/org/elasticsearch/action/bulk/BulkProcessorRetryIT.java

```diff
@@ -0,0 +1,193 @@
+                .put("threadpool.same.queue_size", 1)
+                // don't mess with this one! It's quite sensitive to a low queue size
+                // (see also ThreadedActionListener which is happily spawning threads) even when we already got rejected
+                //.put("threadpool.listener.queue_size", 1)
+                .put("threadpool.get.queue_size", 1)
+                .put("threadpool.search.queue_size", 1)
+                .put("threadpool.suggest.queue_size", 1)
+                .put("threadpool.percolate.queue_size", 1)
+                .put("threadpool.management.queue_size", 1)
+                .put("threadpool.refresh.queue_size", 1)
+                .put("threadpool.warmer.queue_size", 1)
+                .put("threadpool.snapshot.queue_size", 1)
+                .put("threadpool.force_merge.queue_size", 1)
+                .build();
+    }
+
+
+    public void testSimulatedBulkRejectionLoadWithoutBackoff() throws Throwable {
+        int numberOfAsyncOps = randomIntBetween(600, 700);
+        final CountDownLatch latch = new CountDownLatch(numberOfAsyncOps);
+        final Set<Object> responses = Collections.newSetFromMap(new ConcurrentHashMap<>());
+
+        BulkProcessor bulkProcessor = BulkProcessor.builder(client(), new BulkProcessor.Listener() {
+            @Override
+            public void beforeBulk(long executionId, BulkRequest request) {
+                // no op
+            }
+
+            @Override
+            public void afterBulk(long executionId, BulkRequest request, BulkResponse response) {
+                responses.add(response);
+                latch.countDown();
+            }
+
+            @Override
+            public void afterBulk(long executionId, BulkRequest request, Throwable failure) {
+                responses.add(failure);
+                latch.countDown();
+            }
+        }).setBulkActions(1)
+                .setConcurrentRequests(100)
+                .build();
+        indexDocs(bulkProcessor, numberOfAsyncOps);
+        latch.await(10, TimeUnit.SECONDS);
+        bulkProcessor.close();
+
+        assertThat(responses.size(), equalTo(numberOfAsyncOps));
+
+        // validate all responses
+        for (Object response : responses) {
+            if (response instanceof BulkResponse) {
+                BulkResponse bulkResponse = (BulkResponse) response;
+                for (BulkItemResponse bulkItemResponse : bulkResponse.getItems()) {
+                    if (bulkItemResponse.isFailed()) {
+                        BulkItemResponse.Failure failure = bulkItemResponse.getFailure();
+                        Throwable rootCause = ExceptionsHelper.unwrapCause(failure.getCause());
+                        // we're overwhelming the pool, that's the exception we want to get
+                        if (rootCause instanceof EsRejectedExecutionException == false) {
+                            fail("Got unexpected reason [" + failure.getMessage() + "]");
```

Comment:
Well, this is not calling Exception#getMessage() but rather Failure#getMessage() so I guess this will not go into forbidden APIs. And apart from that: It is a test, I know that it is an `EsRejectedExecutionException` and I am just interested in the message...